src/vcFBX.cpp

```diff
@@ -260,6 +260,38 @@ FbxAMatrix vcFBX_GetGeometryTransformation(FbxNode *inNode)
   return FbxAMatrix(lT, lR, lS);
 }
 
+void vcFBX_CleanMaterials(udChunkedArray<vcFBXMaterial> *pMats)
+{
+  // Clean up
+  for (uint32_t i = 0; i < pMats->length; ++i)
+  {
+    for (uint32_t j = 0; j < (*pMats)[i].textures.length; ++j)
+    {
+      if ((*pMats)[i].textures[j].pPixels)
+      {
+        stbi_image_free((*pMats)[i].textures[j].pPixels);
+
+        // Clean up all references to this allocation
+        void *pPtr = (*pMats)[i].textures[j].pPixels;
+        for (uint32_t k = 0; k < pMats->length; ++k)
```

Comment:
This should start at `i`, and `l` should start at `j + 1`. The search only needs to go forwards since you've already destroyed the materials and textures up to `i` and `j`.